nifi-system-tests/nifi-system-test-suite/src/test/java/org/apache/nifi/tests/system/NiFiClientUtil.java

```diff
@@ -397,6 +397,27 @@ public void waitForConnectorValidationStatus(final String connectorId, final Str
         }
     }
 
+    public void waitForConnectorStopped(final String connectorId) throws NiFiClientException, IOException, InterruptedException {
+        waitForConnectorState(connectorId, ConnectorState.STOPPED);
+    }
+
+    public void waitForConnectorState(final String connectorId, final ConnectorState desiredState) throws InterruptedException, NiFiClientException, IOException {
+        int iteration = 0;
+        while (true) {
```

Comment:
This could hang of the condition is never met. Is this expected for these system tests? Assuming so, this comment can be ignored.